main.cpp

```diff
@@ -59,140 +64,452 @@ New/This/Pointers/References conclusion
-    double subtract( double lhs, double rhs );
-    double multiply( double lhs, double rhs );
-    double divide( double lhs, double rhs );
-};
+    *heapFloat -= operand;
+    return *this;
+}
+
+FloatType& FloatType::subtract( const FloatType& operand )
+{
+    *heapFloat -= *operand.heapFloat;
+    return *this;
+}
+
+FloatType& FloatType::subtract( const DoubleType& operand )
+{
+    *heapFloat -= *operand.heapDub;
+    return *this;
+}
+
+FloatType& FloatType::subtract( const IntType& operand )
+{
+    *heapFloat -= *operand.heapInt;
+    return *this;
+}
+
+FloatType& FloatType::multiply( float operand )
+{
+    *heapFloat *= operand;
+    return *this;
+}
+
+FloatType& FloatType::multiply( const FloatType& operand )
+{
+    *heapFloat *= *operand.heapFloat;
+    return *this;
+}
+
+FloatType& FloatType::multiply( const DoubleType& operand )
+{
+    *heapFloat *= *operand.heapDub;
+    return *this;
+}
+
+FloatType& FloatType::multiply( const IntType& operand )
+{
+    *heapFloat *= *operand.heapInt;
+    return *this;
+}
+
+FloatType& FloatType::divide( float operand )
+{
+    *heapFloat /= operand;
+    return *this;
+}
+
+FloatType& FloatType::divide( const FloatType& operand )
+{
+    *heapFloat /= *operand.heapFloat;
+    return *this;
+}
+
+FloatType& FloatType::divide( const DoubleType& operand )
+{
+    *heapFloat /= *operand.heapDub;
+    return *this;
+}
+
+FloatType& FloatType::divide( const IntType& operand )
+{
+    *heapFloat /= *operand.heapInt;
+    return *this;
+}
 
 // double member functions
-double DoubleType::add( double lhs, double rhs )
+DoubleType& DoubleType::add( double operand )
 {
-    return lhs + rhs;
+    *heapDub += operand;
+    return *this;
 }
 
-double DoubleType::subtract( double lhs, double rhs )
+DoubleType& DoubleType::add( const FloatType& operand )
 {
-    return lhs - rhs;
+    *heapDub += *operand.heapFloat;
+    return *this;
 }
 
-double DoubleType::multiply( double lhs, double rhs )
+DoubleType& DoubleType::add( const DoubleType& operand )
 {
-    return lhs * rhs; 
+    *heapDub += *operand.heapDub;
+    return *this;
 }
 
-double DoubleType::divide( double lhs, double rhs )
+DoubleType& DoubleType::add( const IntType& operand )
 {
-    return lhs / rhs;
+    *heapDub += *operand.heapInt;
+    return *this;
 }
 
-// int UDT
-struct IntType
+DoubleType& DoubleType::subtract( double operand )
 {
-    int add( int lhs, int rhs );
-    int subtract( int lhs, int rhs );
-    int multiply( int lhs, int rhs );
-    int divide( int lhs, int rhs );
-};
+    *heapDub -= operand;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract( const FloatType& operand )
+{
+    *heapDub -= *operand.heapFloat;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract( const DoubleType& operand )
+{
+    *heapDub -= *operand.heapDub;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract( const IntType& operand )
+{
+    *heapDub -= *operand.heapInt;
+    return *this;
+}
```

Comment:
revise all of your member functions that take UDTs to call the member function that takes a primitive.